nalgebra-sparse/src/io/matrix_market.rs

```diff
@@ -1329,3 +1429,112 @@ fn next_dense_coordinate(
         }
     }
 }
+
+/// Write a sparse matrix into Matrix Market format string.
+///
+/// Our exporter only writes matrix into `coordiante` and `general` format.
+///
+///
+/// Examples
+/// --------
+/// ```
+/// # use matrixcompare::assert_matrix_eq;
+/// use nalgebra_sparse::io::{write_to_matrix_market_str,load_coo_from_matrix_market_str};
+/// let str = r#"
+/// %%matrixmarket matrix coordinate integer general
+/// 5 4 2
+/// 1 1 10
+/// 2 3 5
+/// "#;
+/// let matrix = load_coo_from_matrix_market_str::<i32>(&str).unwrap();
+/// let generated_matrixmarket_string = write_to_matrix_market_str(&matrix);
+/// // 'generated_matrixmarket' should equal to the 'matrix'
+/// let generated_matrixmarket = load_coo_from_matrix_market_str::<i32>(&generated_matrixmarket_string).unwrap();
+/// assert_matrix_eq!(matrix,generated_matrixmarket);
+/// ```
+pub fn write_to_matrix_market_str<T: MatrixMarketScalar, S: MatrixMarketExport<T>>(
+    sparse_matrix: &S,
+) -> String {
+    let mut bytes = Vec::<u8>::new();
+    // This will call impl<A: Allocator> Write for Vec<u8, A>
+    // The vector will grow as needed.
+    // So, unwrap here won't cause any issue.
+    write_to_matrix_market(&mut bytes, sparse_matrix).unwrap();
+
+    String::from_utf8(bytes)
+        .expect("Unexpected non UTF-8 data was generated when export to matrix market string")
+}
+
+/// Write a sparse matrix into Matrix Market format file.
+///
+/// Our exporter only writes matrix into `coordiante` and `general` format.
+///
+///
+/// Errors
+/// --------
+///
+/// See [MatrixMarketErrorKind] for a list of possible error conditions.
+///
+/// Examples
+/// --------
+/// ```
+/// use nalgebra_sparse::io::{write_to_matrix_market_file,load_coo_from_matrix_market_str};
+/// let str = r#"
+/// %%matrixmarket matrix coordinate integer general
+/// 5 4 2
+/// 1 1 10
+/// 2 3 5
+/// "#;
+/// let matrix = load_coo_from_matrix_market_str::<i32>(&str).unwrap();
+/// let res = write_to_matrix_market_file(&matrix,"path/to/matrix.mtx");
+/// if res.is_err(){
+/// // do something
+/// }
```

Comment:
I think `?` can not be used in doc, because it requires the function signature `fn fun() -> Result<>`, but in the doc, we don't write the function signature, and it's controlled by `cargo test` .
Unless we use `ignore` attribute for this example, otherwise this example can't compile